### DISCUSSION TITLE
Offline on Mobile Only

### DISCUSSION THREAD

**Discussion by JanFeld** [firefly-iii/firefly-iii#5160] [General]
I'm wondering if it's a bug or an invalid configuration on the serverside.. I'm always getting the page "Offline" (image below) when accessing my firefly instance from iOS 15 with Firefox.

My device is online (i'm currently writing this on my phone). The desktop version is working fine! This is the first time i'm observing this, started shortly after upgradimg to 5.6.1. I don't want to open an issue, if it would be something deeper other people would observe this too.

Any ideas? 

![image](https://user-images.githubusercontent.com/22739965/136437200-b3718a66-e5f3-4c88-bf3d-e7bd82046178.jpeg)

**Comment by JC5** [comment]
I'm not sure, maybe @LBreda knows?

**Comment by LBreda** [comment]
It's pretty strange, it seems it can't download the icons font either, so it actually can't reach the server. Could it be a DNS cache issue? Did you change the server address recently?

**Reply by JanFeld** [reply]
I didn't change the server address.
I've noticed when trying with _Safari_ it tells me the Certificate is invalid.

I've checked my certificates (certbot/letsencrypt) and they are valid for another 65 days.
Is there something on iOS that behaves differently than most desktop browsers?
Firefox on desktop as well as iOS tells me the connection is secure.

**Reply by LBreda** [reply]
Do you have some firewall or vpn which may filter the connection using a different certificate? Can you verify who certifies the "invalid" certificate? If it is overwritten by some filter it should not have "Let's Encrypt" as the verifier.

**Reply by JanFeld** [reply]
Not at all. I've tried multiple networks and mobile connections. Are there some changes regrsing the dynamic loading of content? I'm running nginx configured with a reverse proxy and php-fpm. Coule there be something that behaves differently in mobile views? Desktop still works fine.

**Comment by JC5** [comment]
Is there anything in particular I can do to help to debug similar issues?

**Comment by JanFeld** [comment]
Update: even after the update to 5.6.2 it doesn't work. 

I've tried the `/health` on mobile too, same error.
I've tryed renewing and double checked my certificates - no success.
Desktop still works fine - except safari on desktop does the same as the one on iOS.

Invalid/expired certificate - valid until 21. December 2021  ....?

**Reply by JC5** [reply]
Mmm. I also have Firefox and iOS 15. Not a beta, just plain old 15.0.1 (I think). No issues.

Are you running behind lets encrypt?

**Reply by JanFeld** [reply]
Yes, i'm running certbot/letsencrypt. Are there any known issues?

**Reply by JC5** [reply]
No, I'm running the exact same combo :( but I'll do some double-checking, it's been a while since i installed the "app" so perhaps a reinstall of sorts will trigger the same problem

**Reply by JanFeld** [reply]
Thanks for the effort you put into this. I hope it'll sort out someday and is just something strange going on with my configuration. But my other sites hosted on the same machine with the same settings are working fine (gitea for example).

**Reply by JC5** [reply]
Can you install the demo site as an app?

**Reply by JanFeld** [reply]
I'll try that later and let you know!

**Reply by JanFeld** [reply]
No success. Same behaviour like my current installation. I've setup a new instance but couldn't make any progress.

**Reply by JC5** [reply]
Weird. I'm thinking it's something local because I can install the demo site as an app just fine. But I have no clue what it could be.

Do you have any content managers in Safari, like adblockers, iVerify or others?

**Reply by JanFeld** [reply]
Not at all. Maybe it has something to do with the certificates and the 'security' standard apple requires. On Desktop Firefox it's fine, Safari tells me it's insecure. Mobile Safari behaves the same like desktop Safari.. Possibly something with Apples certificate storage and the current certificate i have. I think Apple doesn't trust my Let's encrypt certificates in particular. Strangely it started after the upgrade to 5.6.1. Maybe i should try installing 5.6.0 or any earlier version to verify.

**Reply by JC5** [reply]
There is a certificate root file in the base image that I can update, see if that helps.

https://dev.azure.com/Firefly-III/_git/BaseImage?path=/apache-8.0/Dockerfile&version=GBmain&line=12&lineEnd=13&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

I'm not sure if it's the cause of the issue but it's worth a try.

**Reply by JanFeld** [reply]
Oh, i forgot to mention i'm using the self-hosted approach with composer.. Sorry for that ..

**Reply by JC5** [reply]
Haha no worries, it was a good thing to update it anyway. But this is also me running out of ideas, I'm afraid :(

**Reply by JanFeld** [reply]
I'm sure this isn't a permanent issue. We'll see if it happens after the next due-date for the certificates (in December). Maybe there are some FFIII updates in between too .. ;-) 

Nevertheless, thanks for your help!
